cannon/mipsevm/tests/evm_common_test.go

```diff
@@ -378,6 +378,95 @@ func TestEVMSingleStep_MovzMovn(t *testing.T) {
 
 }
 
+func TestEVMSingleStep_MfhiMflo(t *testing.T) {
+	var tracer *tracing.Hooks
+	versions := GetMipsVersionTestCases(t)
+	cases := []struct {
+		name  string
+		funct uint32
+	}{
+		{name: "mflo", funct: uint32(0x12)},
+		{name: "mfhi", funct: uint32(0x10)},
```

Comment:
(nit) Instead of the if/else switching in in the body of the test, you can just define more test params here:
```suggestion
		{name: "mflo", funct: uint32(0x12), hi: 0x98, lo: 0x76, expectedVal: 0x76},
		{name: "mfhi", funct: uint32(0x10), hi: 0x98, lo: 0x76, expectedVal: 0x98},
```
Then you can always set both hi and low, and always assert `expected.Registers[rdReg] = tt.expectedVal`